opus/application/static_media/js/selectMetadata.js

```diff
@@ -129,16 +131,38 @@ var o_selectMetadata = {
         let buttonTitle = (tab === "#cart" ? "Download CSV (in cart)" : "Download CSV (all results)");
 
         if (!o_selectMetadata.rendered) {
+            let spinnerTimer = setTimeout(function() {
+                $("#op-select-metadata .op-menu-spinner.spinner").addClass("op-show-spinner"); }, opus.spinnerDelay);
+
             // We use getFullHashStr instead of getHash because we want the updated
             // version of widgets= even if the main URL hasn't been updated yet
-            let url = "/opus/__forms/metadata_selector.html?" + o_hash.getFullHashStr();
-
-            // When select metadata contents are outdated, we empty the old contents and display a
-            // small spinner before the new contents are fully loaded.
-            $(".modal-body.op-select-metadata-details").empty();
-            $(".modal-body.op-select-metadata-details").html(opus.spinner);
+            let hash = o_hash.getFullHashStr();
+
+            // Figure out which categories are already expanded
+            let numberCategories = $("#op-select-metadata .op-submenu-category").length;
+            let expandedCategoryLinks = $("#op-select-metadata .op-submenu-category").not(".collapsed");
+            let expandedCategories = [];
+            $.each(expandedCategoryLinks, function(index, linkObj) {
+                expandedCategories.push($(linkObj).data("cat"));
+            });
+            let expandedCats = "";
+            if (numberCategories > 0) {
+                /* If there aren't any categories, it means this is the first time we're
+                   loading the select metadata menu so let the backend do its default
+                   behavior. Otherwise, specify the categories to expand. */
+                if (hash !== "") {
+                    expandedCats = "&";
+                }
+                expandedCats += "expanded_cats=" + expandedCategories.join();
+            }
+            o_selectMetadata.lastMetadataMenuRequestNo++;
+            let url = `/opus/__metadata_selector.json?${hash}${expandedCats}&reqno=${o_selectMetadata.lastMetadataMenuRequestNo}`;
 
-            $(".modal-body.op-select-metadata-details").load(url, function(response, status, xhr)  {
+            $.getJSON(url, function(data) {
+                if (data.reqno < o_menu.lastSearchMenuRequestNo) {
```

Comment:
I think this data.reqno should compare with o_selectMetadata.lastMetadataMenuRequestNo (the one passed in API URL).